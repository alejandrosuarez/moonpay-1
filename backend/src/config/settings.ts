export const COINBASE_EXCHANGE = "coinbase";
export const BINANCE_EXCHANGE = "binance";
export const KRAKEN_EXCHANGE = "kraken";