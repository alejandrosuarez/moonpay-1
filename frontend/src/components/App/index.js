import React, { Component } from "react";
import { connect } from "react-redux";
import Filter from "../Filter";
import Currencies from "../Currencies";
import svg from "../../assets/logo.svg";
import fr from "../../assets/fr.png";
import en from "../../assets/uk.png";
import { Grommet, Button, Avatar } from "grommet";
import "./index.scss";
import { LOCALE_CHANGE } from "../../redux/locale/locale.types";

// Grommet Theme override
const theme = {
  global: {
    focus: {
      border: {
        color: undefined,
      },
    },
  },
  checkBox: {
    color: {
      light: "black",
    },
  },
};

class App extends Component {
  render() {
    return (
      <Grommet plain theme={theme} className="m-app">
        <div className="m-app__layout">
          <div className="m-app__layout__header">
            <img src={svg} alt="logo" />
            <Filter />
          </div>
          <Currencies />
        </div>
        <div className="m-app__locale">
          <Button
            round="full"
            plain={true}
            onClick={() => {
              this.props.localeChange(this.props.locale === "en" ? "fr" : "en");
            }}
          >
            {() => <Avatar src={this.props.locale === "en" ? fr : en} />}
          </Button>
        </div>
      </Grommet>
    );
  }
}

const mapStateToProps = (state) => ({
  locale: state.locale.locale,
});

const mapDispatchToProps = (dispatch) => {
  return {
    localeChange: (locale) =>
      dispatch({
        type: LOCALE_CHANGE,
        payload: locale,
      }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
