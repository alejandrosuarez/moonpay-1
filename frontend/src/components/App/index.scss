@import "../../styles/index.scss";

.m-app {
  display: flex;
  justify-content: center;

  &__layout {
    @include wrapper();
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    &__header {
      display: flex;
      flex-direction: column;
      max-width: rem(500);

      img {
        width: 100%;
      }
    }
  }

  &__locale {
    position: fixed;
    bottom: rem(20);
    right: rem(20);
  }
}
