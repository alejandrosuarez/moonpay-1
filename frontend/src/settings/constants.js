export const LOCALE_DEFAULT = "en";

export const SORT_BY_NAME = "name";
export const SORT_BY_CODE = "code";
export const SORT_BY_RANDOM = "random";

export const FILTER_US = "filter_us";
export const FILTER_TEST_MODE = "filter_test_mode";
export const FILTER_SORT = "filter_sort";
