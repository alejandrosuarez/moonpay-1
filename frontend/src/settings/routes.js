export const ApiRoutes = {
  CURRENCIES: "/currencies",
};

export const InternalRoutes = {};
