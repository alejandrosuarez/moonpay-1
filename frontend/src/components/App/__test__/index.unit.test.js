import React from "react";
import { Provider } from "react-redux";
import App from "../index";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";
import { render, fireEvent, screen } from "../../../settings/test-utils";
import configureStore from "redux-mock-store";

jest.mock("../../Currencies", () => () => <div></div>);
jest.mock("../../Filter", () => () => <div></div>);

const mockStore = configureStore([]);

describe("App Component", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      locale: "en",
      filter: {},
    });

    component = create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  afterEach(cleanup);

  it("Renders without crashing", () => {
    const div = document.createElement("div");
    render(<App />, div);
  });

  it("Matches snapshot empty", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
