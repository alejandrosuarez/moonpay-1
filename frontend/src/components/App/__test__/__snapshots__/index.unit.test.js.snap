// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Matches snapshot empty 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 cMWRON m-app"
>
  <div
    className="m-app__layout"
  >
    <div
      className="m-app__layout__header"
    >
      <img
        alt="logo"
        src={Object {}}
      />
      <div />
    </div>
    <div />
  </div>
  <div
    className="m-app__locale"
  >
    <button
      className="StyledButton-sc-323bzc-0 ebqXLG"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 cfptzR StyledAvatar-sc-1suyamb-1"
      />
    </button>
  </div>
</div>
`;
